http/typescript/fetch/src/fetchOptions.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * @interface {@link https://github.com/bitinn/node-fetch/#options}
+ * Signature to define the fetch request options for node environment
+ * @property {number} [follow] - node-fetch option: maximum redirect count. 0 to not follow redirect
+ * @property {number} [compress] - node-fetch option: support gzip/deflate content encoding. false to disable
+ * @property {number} [size] - node-fetch option: maximum response body size in bytes. 0 to disable
+ * @property {any} [agent] - node-fetch option: HTTP(S).Agent instance, allows custom proxy, certificate, lookup, family etc.
+ * @property {number} [highWaterMark] - node-fetch option: maximum number of bytes to store in the internal buffer before ceasing to read from the underlying resource.
+ * @property {boolean} [insecureHTTPParser] - node-fetch option: use an insecure HTTP parser that accepts invalid HTTP headers when `true`.
+ */
+ export interface NodeFetchInit {
+	follow?: number;
+	compress?: boolean;
+	size?: number;
+	agent?: any;
+	highWaterMark?: number;
+	insecureHTTPParser?: boolean;
+}
+
+/**
+ * @interface
+ * Signature to define the fetch api options which includes both fetch standard options and also the extended node fetch options
+ * @extends RequestInit @see {@link https://fetch.spec.whatwg.org/#requestinit}
+ * @extends NodeFetchInit
+ */
+export interface FetchOptions extends RequestInit, NodeFetchInit {}
```

Comment:
Should we use this interface here ?https://github.com/microsoft/kiota/blob/4f670c3dd5501cfe96b7413187d83e20822977ae/http/typescript/fetch/src/httpClient.ts#L43

Reply:
I cannot think of any default fetch options. What are the options that you think can be set as default?

Reply:
default compression, follow (redirect), timeout, etc....

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c00dae848d207c182dff9a1dceae179e2edff15e/types/node-fetch/index.d.ts#L45

Reply:
Rephrasing my question, what will the default values be here?

Reply:
compression enabled, follow disabled (because handled by a middleware I'm guessing), timeout -> what do we currently have?